File: java_solutions/559.maximum-depth-of-n-ary-tree.java
/*
 * @lc app=leetcode id=559 lang=java
 *
 * [559] Maximum Depth of N-ary Tree
 */

class Solution {
    public int maxDepth(Node root) {
        return root == null ? 0 : 1 + root.children.stream().mapToInt(this::maxDepth).max().orElse(0);
    }
}

File: kotlin_solutions/559.maximum-depth-of-n-ary-tree.kt
/*
 * @lc app=leetcode id=559 lang=kotlin
 *
 * [559] Maximum Depth of N-ary Tree
 */

class Solution { fun maxDepth(root: Node?): Int = if (root == null) 0 else 1 + (root.children.maxOfOrNull { maxDepth(it) } ?: 0) }

